src/KubernetesClient/Extensions.cs

```diff
@@ -0,0 +1,36 @@
+using System;
+using System.Reflection;
+using System.Text.RegularExpressions;
+using k8s.Models;
+
+namespace k8s
+{
+    public static class Extensions
+    {
+
+        public static KubernetesEntityAttribute GetKubernetesTypeMetadata<T>(this T obj) where T : IKubernetesObject =>
```

Comment:
We now have an `IMetadata` interface:

https://github.com/kubernetes-client/csharp/blob/master/src/KubernetesClient/IMetadata.cs

I think it makes sense to use that here.

Reply:
This is different metadata as in this context I mean grab the type's `KubernetesEntityAttribute` instead of Metadata field. I guess I can rename it to something else - I can see the confusion